### ISSUE TITLE
Add integration tests without adding a simapp in this repo

### ISSUE THREAD

**Issue by julienrbrt** [cosmosregistry/example#2]
The boilerplate of this repository should only contain module code.
Right now, we use unit tests for testing the module itself, but we need to as well test the integration with an app.
Instead of adding a complete minimal app in the repository, which results in more code to maintain and more boilerplate that is only required for testing, create a minimal injectable application that can take extra modules.
We can then use that minimal application as a library for testing and not adding direct dependencies to the module while not necessary.

I think this can already be done by using https://github.com/julienrbrt/chain-minimal and [simtestutil](https://pkg.go.dev/github.com/cosmos/cosmos-sdk/testutil/sims).
Basically, we need to import chain-minimal depinject config and append this module depinject config.
Then we can use the runtime.App for testing.

Additionally, we should add a script that builds that app for testing.
